### PR TITLE
Feat/login cb 6 nhat

### PR DESCRIPTION
**Pull request by NhatNguyen1502**

### CODE REVIEW

**Review by ZietMy — APPROVED** [review]

**Review by vthu191 — APPROVED** [review]

File: src/components/ProtectedRoute.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      await fetchInitialSession();
+      if (isMounted) setLoading(false);
+    };
+
+    loadSession();
+
+    return () => {
+      isMounted = false;
+    };
+  }, [setUser]);
```
Comment (khoanguyen-st) [inline src/components/ProtectedRoute.jsx:42]: Blocking: Why use `setUser` for dependency here?

File: src/components/ProtectedRoute.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+   console.log("Current state:", user?.user_metadata);
```
Comment (khoanguyen-st) [inline src/components/ProtectedRoute.jsx:44]: Blocking: Remove log

File: src/components/login/LoginComponent.jsx [hunk 3]
```diff
@@ -1,73 +1,128 @@
...
+
+      const { data, error } = await supabase.auth.signInWithOAuth({
+        provider: "google",
+        options: {
+          redirectTo: "http://localhost:5173/",
+        },
+      });
+
+      if (error) {
+        console.error("Login Error:", error);
```
Comment (khoanguyen-st) [inline src/components/login/LoginComponent.jsx:23]: Blocking: Remove log

File: src/components/login/LoginComponent.jsx [hunk 4]
```diff
@@ -1,73 +1,128 @@
...
-                    viewBox="0 0 24 24" 
-                    fill="#1877F2">
-                    <path 
-                        d="M22.675 0h-21.35C.595 0 0 .593 0 1.326v21.348C0 23.407.595 24 1.325 24h11.495v-9.294H9.69v-3.622h3.13V8.413c0-3.1 1.892-4.788 4.657-4.788 1.324 0 2.462.099 2.794.143v3.24l-1.917.001c-1.504 0-1.795.715-1.795 1.763v2.31h3.587l-.467 3.622h-3.12V24h6.116c.73 0 1.325-.593 1.325-1.326V1.326C24 .593 23.405 0 22.675 0z"
-                    />
-                </svg>
-                <span className="text-gray-900 font-medium"> Sign in with Facebook</span>
-            </button>
-        </div>  
+      console.log("Login Success:", data);
```
Comment (khoanguyen-st) [inline src/components/login/LoginComponent.jsx:28]: Blocking: Same as above

File: src/components/login/LoginComponent.jsx [hunk 5]
```diff
@@ -1,73 +1,128 @@
...
+    } catch (err) {
+      console.error("Unexpected login error:", err);
```
Comment (khoanguyen-st) [inline src/components/login/LoginComponent.jsx:30]: Blocking: Same as above

File: src/components/ProtectedRoute.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          localStorage.removeItem("access_token");
+        }
+      } catch (error) {
+        console.error("Error fetching initial session:", error.message);
+        setUser(null);
+        localStorage.removeItem("access_token");
+      }
+    };
+
+    let isMounted = true;
```
Comment (khoanguyen-st) [inline src/components/ProtectedRoute.jsx:31]: blocking: `isMounted` always true here

**Comment by khoanguyen-st** [comment]
Should create a common component `Toast` to show error to user when needed

File: src/components/ProtectedRoute.jsx (hunk 6 above)
Comment (NhatNguyen1502) [reply src/components/ProtectedRoute.jsx:31]: The block of code below sets isMounted to false to help prevent data leaks when the component is unmounted, right ?

File: src/components/ProtectedRoute.jsx (hunk 1 above)
Comment (NhatNguyen1502) [reply src/components/ProtectedRoute.jsx:42]: I thought it was necessary to handle logout function, but now I realize I was wrong. I only need to clear the data in the logout function. Thank you for pointing that out.

**Pull request merged**